packages/vow/src/watch.js

```diff
@@ -142,33 +117,35 @@ export const prepareWatch = (
   watchPromise = watchPromiseShim,
   rejectionMeansRetry = _reason => false,
 ) => {
-  const watchVow = makeWatchVow(watchPromise);
-  const makeWatcherKit = prepareWatcherKit(zone, rejectionMeansRetry, watchVow);
+  const retryAndWatchVow = makeRetryAndWatchVow(watchPromise);
+  const makePromiseWatcher = preparePromiseWatcher(
+    zone,
+    rejectionMeansRetry,
+    retryAndWatchVow,
+  );
 
   /**
    * @template [T=any]
    * @template [TResult1=T]
    * @template [TResult2=T]
    * @param {import('./types.js').ERef<T | import('./types.js').Vow<T>>} specimenP
    * @param {import('./types.js').Watcher<T, TResult1, TResult2>} [watcher]
+   * @param {unknown} [watcherContext]
    */
-  const watch = (specimenP, watcher) => {
+  const watch = (specimenP, watcher, watcherContext) => {
```

Comment:
We already have a case of calling `watchPromise` with more than one context argument: https://github.com/Agoric/agoric-sdk/blob/18d561f1b95755e58ca691b26938fc249618018d/packages/zoe/src/zoeService/startInstance.js#L100-L102

Such patterns _could_ be converted from `watchPromise(awaitable, watcher, instAdmin, adminNode)` to `watchPromise(awaitable, watcher, [instAdmin, adminNode])` or similar, but that's likely to cost brevity with ESLint and also to be disruptive of the watcher code for transitioning. Do you have a strong reason to prefer a singular `watcherContext` over variadic `...callbackArgs`?

Reply:
> I'm not sure what you mean by "cost brevity with ESLint": does it matter that converting from `watchPromise` to `watch` already saves you 7 letters?

I'm thinking of situations I've encountered where it insists upon line breaks. But that's admittedly not a huge deal either... any signature can work, but you're right that claiming all arguments is a risk. There's just a tradeoff with compatibility. Given the uncertainty, let's leave it for now.